src/Maestro/ReleasePipelineRunner/ReleasePipelineRunner.cs

```diff
@@ -71,6 +71,11 @@ public async Task<TimeSpan> RunAsync(CancellationToken cancellationToken)
             IReliableConcurrentQueue<ReleasePipelineRunnerItem> queue =
                 await StateManager.GetOrAddAsync<IReliableConcurrentQueue<ReleasePipelineRunnerItem>>("queue");
 
+            // Key: BAR BuildId of a build that has completed execution but didn't finish successfully (i.e., the build can be retried).
```

Comment:
This code is getting a little too complex for what it's trying to do.

What do you think of something along this approach?

* Add a retries field to the ReleasePipelineRunnerItem that starts as 0.
* Whenever we dequeue anything, commit the transaction. Removing the item from the queue.
* If there is a failure condition with the queueItem that should be retried:
  * Check retries. If less than maximum determined:
    * Increase retries by 1
    * Spin up a task that waits for <time delta * attempts>, then creates a transaction, and enqueues the item with the updated retries.

That way we don't have to keep that collection of retryInfo, and any kind of failure that we deem retry-able gets retried without having to special case.

Reply:
I like this approach.

Reply:
I like this approach.

Reply:
done